nightingal3/llm-pretraining-behaviours
# Pull model performance metrics from huggingface leaderboard

## Description

This adds a script to pull from the huggingface open LLM leaderboard, as well as validation for the resulting JSON files. The underlying dataset repo is first cloned, and then we search through it to generate the results json in a readable format. I made the general format like this:

```json
{
  "model_name": ,
  "last_updated": ,
 "results": {
    "platform (e.g. eval harness)": {
      "dataset": {
          "num_few_shot_examples": {"acc": .....}
        }
   }
  }
}
```

Note: the official open-llm-leaderboard dataset is [here](https://huggingface.co/datasets/open-llm-leaderboard/results/discussions?status=closed), but there's an issue where it's not possible to load it directly through `datasets`. I also considered scraping from the web interface, but I decided it would be better to use the official dataset files.

**To test:** 
```python
python collect_model_scores.py <model_name>
```
You should see `results_<model_name>.json` in the directory `model_scores`.

Then run validation with `pytest`, everything should pass

## References

- Fixes https://github.com/nightingal3/llm-pretraining-behaviours/issues/46
